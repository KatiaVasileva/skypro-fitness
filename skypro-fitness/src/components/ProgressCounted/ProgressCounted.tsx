type ProgressCountedPropsType = {
    courseId: string | undefined;
    workoutId: string | undefined;
  };

import { useNavigate } from "react-router-dom";

export const ProgressCounted = ({ courseId, workoutId }: ProgressCountedPropsType) => {
  const navigate = useNavigate();

  const handleOnClick = () => {
    navigate("/workout/" + courseId + "/" + workoutId);
  };

  return (
    <div className="w-full h-full overflow-x-hidden fixed z-10 bg-dark-gray/50 top-0 left-0">
      <div className="block w-screen min-h-screen mx-auto my-0">
        <div className="h-screen flex items-center">
          <div className="flex flex-col items-center gap-[30px] bg-white w-[323px] md:w-[426px] h-[278px] shadow-[0px_4px_67px_-12px_rgba(0,0,0,0.13)] mx-auto my-0 px-[40px] py-[40px] rounded-[30px] border-[0.7px] border-solid border-[#d4dbe5]">
            <div className="h-[100px] w-[278px] text-center">
              <p className="text-[28px] md:text-[36px] text-black font-medium mb-7">
                Ваш прогресс засчитан!
              </p>
            </div>
            <img
              src="/img/Check-in-Circle.png"
              className="w-16 h-16 cursor-pointer"
              onClick={handleOnClick}
            />
          </div>
        </div>
      </div>
    </div>
  );
};
