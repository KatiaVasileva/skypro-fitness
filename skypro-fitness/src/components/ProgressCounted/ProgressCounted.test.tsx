/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import * as router from "react-router";
import { BrowserRouter } from "react-router-dom";
import { ProgressCounted } from "./ProgressCounted";
import UserProvider from "../../context/UserContext";
import CoursesProvider from "../../context/CoursesContext";
import WorkoutProvider from "../../context/WorkoutContext";

const navigate = jest.fn();

describe("ProgressCounted component", () => {
  beforeEach(() => {
    jest.spyOn(router, "useNavigate").mockImplementation(() => navigate);
  });

  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <UserProvider>
          <CoursesProvider>
            <WorkoutProvider>
              <ProgressCounted courseId={undefined} workoutId={undefined}/>
            </WorkoutProvider>
          </CoursesProvider>
        </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});