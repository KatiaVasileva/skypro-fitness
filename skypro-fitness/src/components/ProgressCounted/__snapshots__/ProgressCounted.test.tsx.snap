// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressCounted component renders correctly 1`] = `
<div>
  <div
    class="w-full h-full overflow-x-hidden fixed z-10 bg-dark-gray/50 top-0 left-0"
  >
    <div
      class="block w-screen min-h-screen mx-auto my-0"
    >
      <div
        class="h-screen flex items-center"
      >
        <div
          class="flex flex-col items-center gap-[30px] bg-white w-[323px] md:w-[426px] h-[278px] shadow-[0px_4px_67px_-12px_rgba(0,0,0,0.13)] mx-auto my-0 px-[40px] py-[40px] rounded-[30px] border-[0.7px] border-solid border-[#d4dbe5]"
        >
          <div
            class="h-[100px] w-[278px] text-center"
          >
            <p
              class="text-[28px] md:text-[36px] text-black font-medium mb-7"
            >
              Ваш прогресс засчитан!
            </p>
          </div>
          <img
            class="w-16 h-16 cursor-pointer"
            src="/img/Check-in-Circle.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
